camel/functions/web_search.py

```diff
@@ -0,0 +1,250 @@
+            j = min(i + n, len(text))
+        chunks.append(text[i:j])
+        i = j
+    return chunks
+
+
+def single_step_agent(prompt: str) -> str:
+    """Single step agent. Summarise texts or answer a question."""
+
+    assistant_sys_msg = BaseMessage.make_assistant_message(
+        role_name="Assistant",
+        content="You are a helpful assistant.",
+    )
+    agent = camel.agents.ChatAgent(assistant_sys_msg)
+    agent.reset()
+
+    user_msg = BaseMessage.make_user_message(
+        role_name="User",
+        content=prompt,
+    )
+    assistant_response = agent.step(user_msg)
+    if assistant_response.msgs is not None:
+        return assistant_response.msg.content
+    return ""
+
+
+def summarise_text(text: str, query: str) -> str:
+    r"""Summarise the information from the text, base on the query if query is
+    given.
+
+    Args:
+        text (string): Text to summarise.
+        query (string): What information you want.
+
+    Returns:
+        string: Strings with information.
+    """
+    summary_prompt = f"Gather information from this text that relative to " \
+                     f"the question, but do not directly answer " \
+                     f"the question.\nquestion: {query}\ntext "
+    # Max length of each chunk
+    max_len = 3000
+    results = ""
+    chunks = create_chunks(text, max_len)
+    # Summarise
+    for i, chunk in enumerate(chunks, start=1):
+        prompt = summary_prompt + str(i) + ": " + chunk
+        result = single_step_agent(prompt)
+        results += result + "\n"
+
+    # Final summarise
+    final_prompt = f"Here are some summarised texts which split from one " \
+                   f"text, Using the information to " \
+                   f"answer the question: {query}.\n\nText: "
+    prompt = final_prompt + results
+
+    response = single_step_agent(prompt)
+
+    return response
+
+
+def search_web(query: str) -> str:
+    r"""Search webs for information. Input a query, this function will use
+    google search engine search related information from the internet, then
+    return a summarised answer.
+
+    Args:
+        query (string): Question you want to be answered.
+
+    Returns:
+        string: Summarised information from webs.
+    """
+    # Google search will return a list of urls
+    result = search_google(query)
+    answer: str = ""
+    for item in result:
+        url = item.get("URL")
+        # Extract text
+        text = text_extract_from_web(str(url))
+        # Using chatgpt summarise text
+        answer = summarise_text(text, query)
+
+        # Let chatgpt decide whether to continue search or not
+        prompt = f"Do you think the answer: {answer} can answer the query: " \
+                 f"{query}. Use only 'yes' or 'no' to answer."
+        # Add the source
+        answer += f"\nFrom: {url}"
+
+        reply = single_step_agent(prompt)
+        if "yes" in str(reply).lower():
+            break
+
+    return answer
```

Comment:
Should we return something like `"Failed to find the answer from google search."` if all replies are `no`?